### ISSUE TITLE
scrollToIndex out of range error

### ISSUE THREAD

**Issue by heidtjoh** [erksch/react-native-wheely#36]
When scrolling very fast to the top of large lists (example settings: length of options array >= 100, deceleration_rate="normal"), the following error can occur on android phones (tested on different phones): 

Fatal Exception: com.facebook.react.common.JavascriptException
Invariant Violation: scrollToIndex out of range: requested index -1 but minimum is 0

**Comment by c-goettert** [comment]
I am facing the same issue (also on iOS Simulator). I just need to "swipe up" a few times when I reached the first item, then the error occurs.

**Comment by erksch** [comment]
I was also able to reproduce this on iOS. I'll look at it this week.

**Comment by erksch** [comment]
The problem is here:

```typescript
const handleMomentumScrollEnd = (
    event: NativeSyntheticEvent<NativeScrollEvent>,
  ) => {
    const offsetY = event.nativeEvent.contentOffset.y;
    let index = Math.floor(Math.floor(offsetY) / itemHeight);
    const last = Math.floor(offsetY % itemHeight);
    if (last > itemHeight / 2) index++;

    if (index !== selectedIndex) {
      onChange(index);
    }
  };
```

In some cases, especially on iOS `event.nativeEvent.contentOffset.y` is actually negative.
This is properly due to the list's bounciness. At the same time the index might also overshoot in the other direction when scrolling down.

**Comment by erksch** [comment]
Released the fix in 0.6.0
